pyquil/tests/test_operator_estimation.py

```diff
@@ -599,3 +599,54 @@ def test_ratio_variance_array():
     var_b = np.array([0.05, 0.5, 5.0])
     ab_ratio_var = ratio_variance(a, var_a, b, var_b)
     np.testing.assert_allclose(ab_ratio_var, np.array([0.028125, 0.0028125, 0.00028125]))
+
+
+def test_measure_observables_noisy_asymmetric_readout_plus_one():
+    qc = get_qc('9q-qvm')
+    expt1 = ExperimentSetting(TensorProductState(plusX(0)), sX(0))
+    expt2 = ExperimentSetting(TensorProductState(plusY(0)), sY(0))
+    expt3 = ExperimentSetting(TensorProductState(plusZ(0)), sZ(0))
+    p = Program()
+    p.define_noisy_readout(0, p00=0.99, p11=0.80)
+    qubs = [0]
+    tomo_expt = TomographyExperiment(settings=[expt1, expt2, expt3], program=p, qubits=qubs)
+
+    for res in measure_observables(qc, tomo_expt, n_shots=1000,
+                                   readout_symmetrize='exhaustive',
+                                   calibrate_readout='plus-eig'):
+        assert np.isclose(1.0, res.expectation, atol=1.e-1)
+
+
+def test_measure_observables_noisy_asymmetric_readout_minus_one():
+    qc = get_qc('9q-qvm')
+    expt1 = ExperimentSetting(TensorProductState(minusX(0)), sX(0))
+    expt2 = ExperimentSetting(TensorProductState(minusY(0)), sY(0))
+    expt3 = ExperimentSetting(TensorProductState(minusZ(0)), sZ(0))
+    p = Program()
+    p.define_noisy_readout(0, p00=0.99, p11=0.80)
+    qubs = [0]
+    tomo_expt = TomographyExperiment(settings=[expt1, expt2, expt3], program=p, qubits=qubs)
+
+    for res in measure_observables(qc, tomo_expt, n_shots=1000,
+                                   readout_symmetrize='exhaustive',
+                                   calibrate_readout='plus-eig'):
+        assert np.isclose(-1.0, res.expectation, atol=1.e-1)
+
+
+def test_measure_observables_uncalibrated_estimate_noisy_asymmetric_readout_plus_one():
+    qc = get_qc('9q-qvm')
+    expt1 = ExperimentSetting(TensorProductState(plusX(0)), sX(0))
+    expt2 = ExperimentSetting(TensorProductState(plusY(0)), sY(0))
+    expt3 = ExperimentSetting(TensorProductState(plusZ(0)), sZ(0))
+    p = Program()
+    p00, p11 = 0.99, 0.80
+    p.define_noisy_readout(0, p00=p00, p11=p11)
+    qubs = [0]
+    tomo_expt = TomographyExperiment(settings=[expt1, expt2, expt3], program=p, qubits=qubs)
+    expected_symm_error = (p00 + p11) / 2
+    expected_expectation_z_basis = expected_symm_error * (1) + (1 - expected_symm_error) * (-1)
+
+    for res in measure_observables(qc, tomo_expt, n_shots=10000,
+                                   readout_symmetrize='exhaustive',
+                                   calibrate_readout='plus-eig'):
+        assert np.isclose(expected_expectation_z_basis, res.raw_expectation, atol=1.e-1)
```

Comment:
Same comments as above. Let's base the test on many runs (balancing with runtime), but also let's test on 2Q observables and other states.

Reply:
Testing on multiple runs now, but I might reduce the number of experiments to balance for runtime.

Reply:
Reduced runtime.